packages/react-reconciler/src/ReactFiberWorkLoop.old.js

```diff
@@ -1845,7 +1846,7 @@ function commitRootImpl(root, renderPriorityLevel) {
     // The first phase a "before mutation" phase. We use this phase to read the
     // state of the host tree right before we mutate it. This is where
     // getSnapshotBeforeUpdate is called.
-    prepareForCommit(root.containerInfo);
+    prepareForCommit(root.containerInfo, isFiberInsideHiddenOrRemovedTree);
```

Comment:
Does the function get inlined? Passing first class functions around sometimes deopts closure. Although mostly if you use a closure which this isn’t.

Reply:
How else can we pass it to the renderer without too much indirection? I intentionally wanted to keep it out so we didn't leak details.

Reply:
I think you can just use `isFiberInsideHiddenOrRemovedTree` directly from the renderer? Kind of not ideal but ReactDOM already does this in a few places.

Reply:
I asked @sebmarkbage about this and he suggested not doing that. Maybe we should here though, just to make it more optimal. I’ll defer for now.